models/topic.go

```diff
@@ -70,6 +97,52 @@ func GetTopicByName(name string) (*Topic, error) {
 	return &topic, nil
 }
 
+// addTopicByNameToRepo adds a topic name to a repo and increments the topic count.
+// Returns topic after the addition
+func addTopicByNameToRepo(repoID int64, topicName string, e Engine) (*Topic, error) {
```

Comment:
Can you set e Engine as first param ? (commonly done across models func)
```suggestion
func addTopicByNameToRepo(e Engine, repoID int64, topicName string) (*Topic, error) {
```